osu.Game/Rulesets/UI/DrawableRuleset.cs

```diff
@@ -196,13 +201,30 @@ private void load(OsuConfigManager config, CancellationToken? cancellationToken)
                 Overlays = new Container { RelativeSizeAxes = Axes.Both }
             };
 
+            var resumeInputManager = CreateInputManager();
+
             if ((ResumeOverlay = CreateResumeOverlay()) != null)
             {
-                AddInternal(CreateInputManager()
+                AddInternal(resumeInputManager
                     .WithChild(CreatePlayfieldAdjustmentContainer()
                         .WithChild(ResumeOverlay)));
             }
 
+            if (RuntimeInfo.IsMobile)
+            {
+                if (!(KeyBindingInputManager is IHasVirtualHandler virtualInputManager))
+                    throw new InvalidOperationException($"A {nameof(KeyBindingInputManager)} which supports virtual input handling is not available");
+
+                virtualInputManager.VirtualInputHandler = CreateVirtualInputHandler();
+                resumeInputManager.UseParentInput = false;
+
+                if (!ProvidingUserCursor)
+                {
+                    // The cursor is hidden by default (see Playfield.load()), but should be shown when there's a replay
```

Comment:
this feels very weird/hacky.

Reply:
I can't really think of a proposal that isn't hacky to fix this, have any?